src/views/ApartmentApp/index.js

```diff
@@ -27,36 +27,32 @@ const ApartApp = ({ authentication }) => {
   const isOnline = useNetworkStatus();
 
   useEffect(() => {
-    let isSubscribed = true;
-    if (authentication) {
+    const loadPage = async () => {
       setLoading(true);
       setIsAuthenticated(true);
+      try {
+        const profileInfo = await user.getProfileInfo();
+        setUserProfile(profileInfo);
+        setIsUserStudent(profileInfo.PersonType.includes('stu'));
+        try {
+          setCanUseStaff(await housing.checkHousingAdmin());
+        } catch {
+          setCanUseStaff(false);
+        } finally {
+          setLoading(false);
```

Comment:
I don't think you need this `finally` branch, since the outer `try` block has a `finally` branch that also sets loading to false. That one will always execute after this one, since this branch is contained inside the try.